test/metabase/api/notification_test.clj

```diff
@@ -661,31 +703,132 @@
+                                                                                                   :user_id (mt/user->id :lucky)}]}]}]
+              ;; Unsubscribe from first notification
+              (mt/user-http-request :lucky :post 200 (format "notification/%d/unsubscribe" noti-1))
 
-            ;; Check first notification has no recipients
+              ;; Check first notification has no recipients
               ;; First notification should have no recipients
-            (is (empty? (email-recipients noti-1)))
+              (is (empty? (email-recipients noti-1)))
               ;; Second notification should still have lucky as recipient
-            (is (=? [{:type    :notification-recipient/user
-                      :user_id (mt/user->id :lucky)}]
-                    (email-recipients noti-2)))))))))
+              (is (=? [{:type    :notification-recipient/user
+                        :user_id (mt/user->id :lucky)}]
+                      (email-recipients noti-2))))))))))
+
+(deftest unsubscribe-receive-email-test
+  (testing "test that unsubscribe will not receive email"
+    (mt/with-model-cleanup [:model/Notification]
+      (notification.tu/with-channel-fixtures [:channel/email]
+        (notification.tu/with-card-notification [{noti-1 :id
+                                                  :as notification} {:notification {:creator_id (mt/user->id :rasta)}
+                                                                     :card         {:name "My Card"}
+                                                                     :handlers     [{:channel_type "channel/email"
+                                                                                     :recipients   [{:type    :notification-recipient/user
+                                                                                                     :user_id (mt/user->id :lucky)}]}]}]
+          (let [[email] (notification.tu/with-mock-inbox-email!
+                          (with-send-messages-sync!
+                            (mt/user-http-request :lucky :post 200 (format "notification/%d/unsubscribe" noti-1))))
+                a-href (format "<a href=\"https://metabase.com/testmb/question/%d\">My Card</a>."
+                               (-> notification :payload :card_id))]
+            (testing "sends unsubscribe confirmation email"
+              (is (=? {:bcc     #{"lucky@metabase.com"}
+                       :subject "You unsubscribed from an alert"
+                       :body    [{"You’re no longer receiving alerts about" true
+                                  a-href                                    true}]}
+                      (mt/summarize-multipart-single-email email
+                                                           #"You’re no longer receiving alerts about"
+                                                           (re-pattern a-href)))))))))))
```

Comment:
Might be worth it to `Pattern/quote` the link, so this test won't break if links include regex symbols in future?